QueryGenerator.php

```diff
@@ -449,6 +449,31 @@ private function getGlue($method) {
          return ") OR (";
       }
    }
+
+   /**
+    * Debug method to print out the full query with the params inserted at the correct locations
+    */
+   public function debugStringifyQuery(): string {
+      $query = $this->build(true);
+      $params = $query[1];
+      $paramValues = array_reduce($params, function ($carry, $param) {
+         static $isInt = false;
+
+         if ($param === 'i') {
+             $isInt = true;
+         } elseif ($param === 's') {
+             $isInt = false;
+         } else {
+             $carry[] = $isInt ? (int)$param : "'$param'";
+         }
```

Comment:
This looks like it shouldn't be in the `else`, but rather below the conditional entirely.

Reply:
This is correct, I'm just doing it in a weird way. We send types, values in the $param array, so we end up with param array that could look like this: 
`["i",1,"i",1,"i",1,"i",1,"i",1,"s","TOPIC","i",1,"s","en","i",1,"i",16,"s","en","i",1]`.

The else captures when we don't match `'i'` or `'s'` and concludes that its a value that we can add to the $carry array.

I'm currently working on expanding this to use all of our DB types rather than just int and string. As part of that, I added a separate check for `$index % 2 == 0` because the types will always be at even indices rather than using this format.